File: ielts-vocab-master/Python/Flask/auth/check_email.py
import re

from flask import Blueprint, jsonify, request

from db import get_db

check_email_bp = Blueprint('check_email', __name__)


@check_email_bp.route('/check_email', methods=['POST'])
def check_email():
    body  = request.get_json(silent=True) or {}
    email = body.get('email', '').strip().lower()

    if not re.match(r'^[^@\s]+@[^@\s]+\.[^@\s]+$', email):
        return jsonify({'error': 'Please enter a valid email address.'}), 400

    conn = get_db()
    row  = conn.execute('SELECT id FROM users WHERE email = ?', (email,)).fetchone()
    return jsonify({'exists': row is not None})

File: IELTS-Vocab/PHP/auth/check_email.php
<?php
header('Content-Type: application/json');
require __DIR__ . '/db.php';

$body = json_decode(file_get_contents('php://input'), true);
$email = isset($body['email']) ? trim(strtolower($body['email'])) : '';

if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
    http_response_code(400);
    echo json_encode(['error' => 'Please enter a valid email address.']);
    exit;
}

$pdo = get_db();
$stmt = $pdo->prepare('SELECT id FROM users WHERE email = ?');
$stmt->execute([$email]);
echo json_encode(['exists' => (bool) $stmt->fetch()]);
